### PR TITLE
BUG: Don't segfault to_numeric when input is empty

### LINKED ISSUE #16302: to_numeric segmentation fault with empty object series
#### Code Sample

```python
from pandas import Series, to_numeric
data = Series(dtype=object)
to_numeric(data, errors='coerce', downcast='integer')

```

Further tracing seems to indicate the problem is likely here:

```
pd._libs.maybe_convert_numeric(data, set(), coerce_numeric=True)

```
#### Problem description

Running to_numeric with an empty object series causes a segmentation fault on version 0.20.1. This did not occur in version 0.19.0.

#### Expected Output

An empty numeric series, or a ValueError.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.1.final.0
python-bits: 64
OS: Darwin
OS-release: 15.0.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.1
pytest: None
pip: 9.0.1
setuptools: 20.10.1
Cython: None
numpy: 1.10.4
scipy: None
xarray: None
IPython: 4.2.0
sphinx: None
patsy: None
dateutil: 2.4.2
pytz: 2015.7
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.0.0b1
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: 0.9.6
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8
s3fs: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

Closes #16302.

### CODE REVIEW

**Review by jorisvandenbossche — APPROVED** [review]

File: pandas/_libs/src/inference.pyx [hunk 1]
```diff
@@ -947,8 +947,13 @@ def maybe_convert_numeric(ndarray[object] values, set na_values,
     -------
     numeric_array : array of converted object values to numerical ones
     """
+
+    if len(values) == 0:
+        return np.array([], dtype='i8')
```
Comment (jreback) [inline pandas/_libs/src/inference.pyx]: this is not correct if the input is an arry, then we would want to return the same dtype (empty).

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by gfyoung** [comment]
@jreback : Everything is green and ready to go.

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**